### PR TITLE
Fix spell check

### PR DESCRIPTION
**Pull request by VictoriousRaptor**
Fix the broken spell check workflow by using the pre-release version of spell check and with some small fixes.

Ref: https://github.com/check-spelling/check-spelling/issues/59

### CODE REVIEW

File: .github/actions/spelling/patterns.txt [hunk 1]
```diff
@@ -118,3 +118,8 @@
 
 # UWP
 [Uu][Ww][Pp]
+
+(?:(?<=[A-Z]{2})V|(?<=[a-z]{2}|[A-Z]{2})v)\d+(?:\b|(?=[a-zA-Z_]))
+
+# version suffix <word>v#
+(?:(?<=[A-Z]{2})V|(?<=[a-z]{2}|[A-Z]{2})v)\d+(?:\b|(?=[a-zA-Z_]))
```
Comment (taooceros) [inline .github/actions/spelling/patterns.txt:125]: What are these two and isn't they the same?

Comment (VictoriousRaptor) [reply .github/actions/spelling/patterns.txt:123]: Sorry about that. It matches string that ends with `V+number`, like `JsonRPCV2`.

Comment (taooceros) [reply .github/actions/spelling/patterns.txt:123]: It's not clear to me why it requires such a complicated pattern? Could you somehow explain?

Comment (VictoriousRaptor) [reply .github/actions/spelling/patterns.txt:123]: It's suggested by the spell check action in my test runs. 58 matches in our repo. Feel free to remove it. It's not vital. Explaination by copliot:

> The given regular expression (?:(?<=[A-Z]{2})V|(?<=[a-z]{2}|[A-Z]{2})v)\d+(?:\b|(?=[a-zA-Z_])) can be broken down as follows:

>(?:(?<=[A-Z]{2})V|(?<=[a-z]{2}|[A-Z]{2})v): This part matches either an uppercase 'V' preceded by two uppercase letters or a lowercase 'v' preceded by two lowercase letters or two uppercase letters. The (?<=...) is a positive lookbehind assertion that checks for the preceding pattern without including it in the match.

>\d+: This part matches one or more digits.

>(?:\b|(?=[a-zA-Z_])): This part matches either a word boundary (\b) or a position followed by a letter (either lowercase or uppercase) or an underscore.

>In summary, this regular expression is used to match a pattern where either an uppercase 'V' or a lowercase 'v' is followed by one or more digits, and it is either preceded by two uppercase letters or two lowercase letters or two uppercase letters followed by a lowercase letter. The match is also followed by a word boundary or a position followed by a letter or an underscore.

Comment (taooceros) [reply .github/actions/spelling/patterns.txt:123]: Ok thanks

**Review by taooceros — APPROVED** [review]

**Pull request merged**